catalog/dags/common/ingestion_server.py

```diff
@@ -24,7 +24,10 @@ def response_filter_stat(response: Response) -> str:
     This is used to extract the name of the current index that the concerned alias
     points to. This index name will be available via XCom in the downstream tasks.
     """
-    index_name = response.json()["alt_names"]
+    data = response.json()
+    if not (data["exists"]):
+        raise AirflowSkipException("Index does not exist.")
```

Comment:
Does this exception cause the entire DAG to skip or are DAGs able to handle skipped tasks individually? If it cannot be individually handled by DAGs (in order to continue if there are expected cases where the index does not exist) then returning None might be better? I don't know if that makes sense in an Airflow context though, so not blocking on this.

Reply:
Normally, skips do propagate downstream! However, Airflow has a few [trigger rules](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#concepts-trigger-rules) which allow you to control the flow of skips and determine how/if some steps should run due to upstream failures or skips. You can see from the "**After**" screenshot in the description what a skipped run looks like (only the `get_current_index` and `trigger_delete_index` get skipped).

A skip is preferable here because:
- It's more visible from the UI that this and affected steps _did not run_ when they get skipped (since they're a different color). That makes it clear at-a-glance what steps actually did run. We had a case earlier (I looked but couldn't find the exact change) where loading steps were completing successfully even if they didn't load anything. It was really hard to tell when those cases where occurring amidst dozens of runs when all of them were green in the grid UI. We changed that to skip and it was immediately much clearer!
- Skipped tasks do not run, so in this case we wouldn't need to add any extra logic to the downstream tasks for how to appropriately handle a `None` value. We can run on the assumption that if a task gets run, it should be receiving a valid value.

Reply:
Thanks for the explanation. Skip makes sense, especially after learning about the "gather" step that comes after this one.